addons/mail/models/mail_thread.py

```diff
@@ -4178,6 +4181,20 @@ def _get_mail_thread_data(self, request_list):
             ])._format_for_chatter()
             res['selfFollower'] = self_follower[0] if len(self_follower) > 0 else None
             res['followers'] = self.message_get_followers()
+            subtype_id = self.env['ir.model.data']._xmlid_to_res_id('mail.mt_comment')
+            self_recipient = self.env['mail.followers'].search([
+                ("res_id", "=", self.id),
+                ("res_model", "=", self._name),
+                ('partner_id', '=', self.env.user.partner_id.id),
+                ('subtype_ids', '=', subtype_id),
+            ])._format_for_chatter()
+            res['selfRecipient'] = self_recipient[0] if len(self_recipient) > 0 else None
+            res['recipientsCount'] = self.env['mail.followers'].search_count([
+                ("res_id", "=", self.id),
+                ("res_model", "=", self._name),
+                ('subtype_ids', '=', subtype_id),
+            ])
+            res['recipients'] = self.message_get_followers(recipients=True)
```

Comment:
We should remove `self` from everywhere, we never send a message to `self` and even if we did, it would not be useful to display it on that list.

Ensure fetch and count query exclude `self`.